model_one/cli.py

```diff
@@ -1,198 +1,274 @@
+        "models": ("GET", f"{URL}/v0/models"),
+        "generate": ("POST", f"{URL}/v0/{{}}/generate"),
+        "classify": ("POST", f"{URL}/v0/{{}}/classify"),
+        "upload": ("POST", f"{URL}/v0/{{}}/upload"),
+        "status": ("POST", f"{URL}/v0/{{}}/status"),
+        "fit": ("POST", f"{URL}/v0/{{}}/fit"),
+    }
 
-    def fit(self):
-        return self._api_post_request('fit', '')
+    @classmethod
+    def _request(cls, method: str, url: str, params: Optional[dict] = None, data: Optional[dict] = None) -> dict:
+        headers = {"Authorization": API_KEY,}
+
+        if data is not None:
+            headers["Content-Type"] = "application/json"
+
+        r = requests.request(method, url, params, data, headers)
 
-    def is_ready(self):
-        return self._status.lower() == 'ready'
-
-    def upload_arrays(self, train_X, train_y, validate_X, validate_y):
-        def build_data(self):
-            data = {
-                'train_dataset': {
-                    'inputs': train_X,
-                },
-                'validate_dataset': {
-                    'inputs': validate_X,
-                }
-            }
-            if self._model_type == 'generator':
-                data['train_dataset']['outputs'] = train_y
-                data['validate_dataset']['outputs'] = validate_y
-            elif self._model_type == 'classifier':
-                data['train_dataset']['classes'] = train_y
-                data['validate_dataset']['classes'] = validate_y
-            else:
-                raise Exception('Unknown model type.')
-            return json.dumps(data)
-        magic_const = 32
-        if len(train_X) < magic_const or len(train_X) < magic_const or len(train_X) < magic_const or len(train_X) < magic_const:
-            raise Exception(
-                'Dataset must be contain at least {} elements'.format(magic_const))
-        return self._api_post_request('upload', build_data(self))
-
-    def upload(self, train_X, train_y, validate_X, validate_y):
-        from pandas import Series
-        if not (isinstance(train_X, Series) and isinstance(train_X, Series)
-                and isinstance(train_X, Series) and isinstance(train_X, Series)):
-            raise Exception('Try upload_arrays.')
-
-        return self.upload_arrays(train_X.tolist(), train_y.tolist(), validate_X.tolist(), validate_y.tolist())
-
-    def _inference(self, method, input):
-        if not input:
-            raise Exception('prompt must not be empty')
-
-        r = requests.get('{}/{}/{}?input={}'.format(_models_api(),
-                         self._id, method, urllib.parse.quote(input)), headers=_get_auth_header())
         if r.status_code != 200:
-            raise Exception(r.text)
+            raise BeyondmlModelException(r.text)
+
         return r.json()
 
-    def generate(self, input):
-        res = self._inference(method='generate', input=input)
-        return res['answer']['responses'][0]['response']
+    @classmethod
```

Comment:
Technically, we could have generated all the methods depending on the contents of V0 dict, but that would require work that is not really rewarding while there are just a few of endpoints